flips/20211007-transaction-fees.md

```diff
@@ -55,151 +57,152 @@ From the users perspective the transaction fees implementation should satisfy th
 
-If the transaction computation limit is reached during the execution of the transaction the execution stops and the state changes made up to that point are dropped. The transaction fees for this transaction are still collected.
+If the transaction execution effort limit is reached during the execution of the transaction the execution stops and the state changes made up to that point are dropped. The transaction fees for this transaction are still collected at the maximum execution effort limit.
 
-Currently the dynamic computation effort is calculated by adding 1 for every loop or function call. This could  be sufficient for the first implementation. In the future the calculation of the dynamic computation effort would be changed so that function calls would have a different computational effort assigned to them according to the time complexity of the method.
+Currently the execution effort is calculated by adding 1 for every loop or function call. This could be sufficient for the first implementation. In the future the calculation of the execution effort would be changed so that function calls would have a different execution effort cost assigned to them according to the time complexity of the method.
 
 ### Failing transactions
 
 There are a few reasons why a transaction fails, which can be split into four categories, according to who finally gets charged for the transaction fees and how much do they get charged.
 
-1. Transaction failed because the payers signature was incorrect, or the payer does not have enough funds to cover maximum transaction costs (transaction costs with the dynamic computation effort set to the computation limit).
+1. Transaction failed because the payers signature was incorrect, or the payer does not have enough funds to cover maximum transaction costs (transaction costs with the execution effort set to the execution effort limit).
 2. Transaction failed before the transaction script execution started (some non-payer signature was incorrect, or the sequence number was incorrect)
 3. Transaction failed during transaction script execution, during fee deduction or during the storage used check. 
-4. Transaction failed because the computation limit was reached.
+4. Transaction failed because the execution effort limit was reached.
 
-As discussed below, in scenario 1. the transaction shouldn't have been included in a collection in the first place. However since it was included and there is no way to charge the payer, the collection node account is charged with the transaction fees. The transaction fees are computed with the dynamic computation effort set to 0.
+As discussed below, in scenario 1. the transaction shouldn't have passed the access node in the first place. However since it has and there is no way to charge the payer, the access node account is charged with the transaction fees. The transaction fees are computed with the execution effort set to 0.
```

Comment:
I'd probably merge the proposal on the charges with the cases above, otherwise I am jumping back and forth between the different cases and  how we propose to charge for them.

Reply:
I switched the order of the chapters, so at the point when you are reading about the categorisation of failing transactions, you should already have knowledge about some transactions not being included due to the payer being unable to pay.

Reply:
Flagging that we should discuss whether or not this is managed at AN or LN (collection node) level. I think currently we're trending toward AN but I do think we should complete the discussion on how failing txs without payer are managed between those 2 node types. I know @AlexHentschel has many thoughts ;)